Framework/Avoidance/Handlers/FireChainsAvoidanceHandler.cs

```diff
@@ -1,29 +1,50 @@
 ﻿using System.Linq;
+using Trinity.Framework.Actors.ActorTypes;
 using Trinity.Framework.Avoidance.Structures;
 using Trinity.Framework.Grid;
+using Zeta.Bot.Navigation;
 
 namespace Trinity.Framework.Avoidance.Handlers
 {
     public class FireChainsAvoidanceHandler : IAvoidanceHandler
     {
-        public void UpdateNodes(TrinityGrid grid, Structures.Avoidance avoidance)
+        public bool UpdateNodes(TrinityGrid grid, Structures.Avoidance avoidance)
         {
-            var actor = avoidance.Actors.FirstOrDefault();
-            if (actor == null)
-                return;
+            var actor = Core.Actors.RactorByRactorId<TrinityActor>(avoidance.RActorId);
+            if (actor == null || !actor.IsValid)
+                return false;
 
-            foreach (var otherAvoidance in Core.Avoidance.CurrentAvoidances)
+            var fireChainFriendList = Core.Avoidance.CurrentAvoidances.Where(c => c != avoidance && c.ActorSno == avoidance.ActorSno)
+                .ToList();
+
+            var appliedAvoidanceNode = false;
+            foreach (var fireChainFriend in fireChainFriendList)
             {
-                if (otherAvoidance == avoidance)
+                var friend = Core.Actors.RactorByRactorId<TrinityActor>(fireChainFriend.RActorId);
+                if (friend == null || !friend.IsValid)
                     continue;
 
-                var fireChainFriend = otherAvoidance.Actors.FirstOrDefault(a => a.ActorSnoId == actor.ActorSnoId);
-                if (fireChainFriend != null)
+                var nodes = grid.GetRayLineAsNodes(actor.Position, friend.Position).SelectMany(n => n.AdjacentNodes);
+
+                if (avoidance.Settings.Prioritize)
+                {
+                    appliedAvoidanceNode = true;
+                    grid.FlagAvoidanceNodes(nodes, AvoidanceFlags.Avoidance | AvoidanceFlags.CriticalAvoidance, avoidance, 50);
+                }
+                else
                 {
-                    var nodes = grid.GetRayLineAsNodes(actor.Position, fireChainFriend.Position).SelectMany(n => n.AdjacentNodes);
+                    appliedAvoidanceNode = true;

```

Comment:
This bool looks redundant now as it is being set to true for all cases

Reply:
No it isn't. There is a continue in a foreach() which may happen and in that case we do not want to update the cell weighting obstacle to the grid.